### ISSUE TITLE
how to get the results

### ISSUE THREAD

**Issue by wwpww** [rainerkelz/ICASSP19#1]
```
onnp_0.9_offp_0.1_thrs_0.5
0.7954    0.773       0.7836     0.8759
0.5763    0.5315     0.5516     0.8631

onnp_0.9_offp_0.9_thrs_0.9
0.8099     0.703        0.7504     0.8759
0.6036     0.4132     0.4861      0.8615
```
I can't get the results like the paper.
I don't know whether I'm wrong in some steps.
Could you tell me how to fix the script in some parameters to get the results and other results about offsets please?
Thank you very much!

**Comment by rainerkelz** [comment]
hi!

you write that you do "not know whether you are wrong in some steps."
so i have the following questions:
did you follow all the steps in the description?
did you let the network train long enough?
how does your loss curve look like?

it might be that you need to adjust the grid search parameters a bit.

in the lines commented out in
https://github.com/rainerkelz/ICASSP19/blob/15b756692cca80e7b16cdbdf28d40f43db2c03b6/gridsearch.py#L109

there are some values for the three parameters that govern the behavior of the simple HMM that postprocesses the network outputs. how fine a grid have you been searching?

i am interpreting the numbers you write as [precision, recall, fmeasure, overlap] as reported by the mir_eval library.
i see that you are close to the numbers in the paper with

"onnp_0.9_offp_0.1_thrs_0.5"

which is only one percentage point off, which i'd say is mainly because your precision is low. i'd say you center your new gridsearch around these parameters, and continue searching. each network needs HMM parameters tuned to its **actual** output probabilities. those are going to be slightly different each time you initialize randomly, and therefore you have to adapt the HMM parameters to them. there's certainly other ways to do this, but we took the easy way out, and did a mixture of manual tuning and gridsearch...

**Comment by wwpww** [comment]
I'm sorry too late.
I followed all the step in your description and trained about 25h on my RTX2080Ti.
![train](https://user-images.githubusercontent.com/48265203/95328009-7f336e80-08d7-11eb-832a-ba308b180754.png)
![valid_best](https://user-images.githubusercontent.com/48265203/95328023-835f8c00-08d7-11eb-9665-e07db6f28533.png)
![valid_f](https://user-images.githubusercontent.com/48265203/95328040-88bcd680-08d7-11eb-81cb-b97466184568.png)
![valid_full_8_1_4](https://user-images.githubusercontent.com/48265203/95328045-8b1f3080-08d7-11eb-9118-85d4b76393f1.png)
![valid_loss](https://user-images.githubusercontent.com/48265203/95328055-8eb2b780-08d7-11eb-9ec7-f0bbf394bbef.png)
![valid_means](https://user-images.githubusercontent.com/48265203/95328061-91151180-08d7-11eb-9b44-eed9ab50c9bd.png)
![valid_on_814](https://user-images.githubusercontent.com/48265203/95328066-92ded500-08d7-11eb-86b9-6df412aed98f.png)
![valid_p](https://user-images.githubusercontent.com/48265203/95328074-95d9c580-08d7-11eb-92a4-0836943a238c.png)
![valid_r](https://user-images.githubusercontent.com/48265203/95328085-98d4b600-08d7-11eb-8592-cf904b25582d.png)

with grindsearch 0.8 0.1 0.4
0.8113 0.7807 0.7949 0.8747-train
0.5867 0.5345 0.5576 0.8627-test
yes, I need to keep searching for the best result.
I just want to know about my loss curve, it seems something is wrong.
Looking forward for your response, thank you!

**Comment by rainerkelz** [comment]
hi, it looks a bit bumpy, yes. maybe you could try and decrease the learning rate a tiny bit?

**Comment by wwpww** [comment]
ok,I'll try, thanks!

**Comment by wwpww** [comment]
There were some warnings when I ran gridsearch and test.
`adsr.py:132: RuntimeWarning: divide by zero encountered in log return np.log(densities)`
Do you think this will affect the results？
